### PR TITLE
Delivery address

### PR DESCRIPTION
**Pull request by San88Francisco**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -29,9 +29,9 @@
     "@typescript-eslint/no-non-null-assertion": "off",
     "arrow-body-style": "off",
     "@typescript-eslint/no-non-null-asserted-optional-chain": "off",
-    "complexity": ["error", 5],
+   //  "complexity": ["error", 5],
     "curly": ["error", "multi-line"],
-    "max-lines": ["error", 255],
+   //  "max-lines": ["error", 255],
```
Comment (pqoqubbw) [inline .eslintrc]: you should disable this rule for file, not for all project

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/components/Account-panel/Delivery-address/Delivery-address.tsx [hunk 2]
```diff
@@ -1,3 +1,233 @@
...
+
+  return {
+    icon,
+    text: type,
+    addressHeadline,
+    location,
+    titleAddress,
+    address_warehouse: addressWarehouse
+  }
+}
```
Comment (pqoqubbw) [inline src/components/Account-panel/Delivery-address/Delivery-address.tsx]: try to do cleaner logic cause now it looks like a mess. try not use let, don't repeat logic and make it more understandable

File: src/components/Account-panel/Delivery-address/Delivery-address.tsx [hunk 3]
```diff
@@ -1,3 +1,233 @@
...
+  const [snackbarData, setSnackbarData] = useState<ISnackbarData>({
+    open: false,
+    error: false
+  })
+  const handleCloseSnackbar = () => {
+    setSnackbarData({ ...snackbarData, open: false })
+  }
+  const handleDeleteClick = (id: number, source: string) => {
+    let url = ''
+    let dataId = {}
```
Comment (pqoqubbw) [inline src/components/Account-panel/Delivery-address/Delivery-address.tsx]: don't use let

File: src/components/Account-panel/Delivery-address/Delivery-address.tsx [hunk 4]
```diff
@@ -1,3 +1,233 @@
...
+      })
+      .catch((error) => {
+        console.error('Error', error)
+        setSnackbarData({ open: true, error: true, message: 'Сталась помилка' })
+      })
+  }
+
+  useEffect(() => {
+    fetchDataMyPostOffices(setDeliveryAddresses)
+  }, [setDeliveryAddresses])
```
Comment (pqoqubbw) [inline src/components/Account-panel/Delivery-address/Delivery-address.tsx]: why do you have setter as a dependency?

File: src/components/Account-panel/Delivery-address/Delivery-address.tsx [hunk 5]
```diff
@@ -1,3 +1,233 @@
...
+    deliveryAddresses.nova_poshta.map((item: AddressItem) => ({
+      ...item,
+      source: 'nova_poshta',
+      addressType: item.address_warehouse ? 'відділення' : 'адреса'
+    })) || []
+  const ukrPoshtaArray: AddressArray =
+    deliveryAddresses.ukr_poshta.map((item: AddressItem) => ({
+      ...item,
+      source: 'ukr_poshta'
+    })) || []
```
Comment (pqoqubbw) [inline src/components/Account-panel/Delivery-address/Delivery-address.tsx]: why do you need that?

File: src/components/Account-panel/Delivery-address/fatchDataPostOffices.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import axiosInstance from '../../../axios/settings'
+
+const fetchDataMyPostOffices = (
+  setDeliveryAddresses: React.Dispatch<
+    React.SetStateAction<{ nova_poshta: never[]; ukr_poshta: never[] }>
```
Comment (pqoqubbw) [inline src/components/Account-panel/Delivery-address/fatchDataPostOffices.ts]: why do you need never here?

File: src/components/AddressForm/AddressForm.tsx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  city_urk?: string
+  postalCode_urk?: string
+  street_urk?: string
+  house_urk?: string
+  apartment_urk?: string
+}
+type AddressFormProps = {
+  setOpenModal: (value: boolean) => void
+  setDeliveryAddresses: React.Dispatch<
+    React.SetStateAction<{ nova_poshta: never[]; ukr_poshta: never[] }>
```
Comment (pqoqubbw) [inline src/components/AddressForm/AddressForm.tsx]: same here

File: src/components/AddressForm/AddressForm.tsx [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+    watch,
+    setValue,
+    formState: { errors }
+  } = useForm<FormValue>({
+    resolver: yupResolver(AddAddressSchema(selectedValue))
+  })
+
+  const onSubmit = (data: FormValue) => {
+    let dataResout
+    let url = ''
```
Comment (pqoqubbw) [inline src/components/AddressForm/AddressForm.tsx]: don't use let

File: src/components/AddressForm/AddressForm.tsx [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    switch (selectedValue) {
+      case 'np_office':
+        dataResout = {
+          city: data.city_np_office,
+          address_warehouse: data.separation_np_office
+        }
+        console.log('✌️dataResout --->', dataResout)
```
Comment (pqoqubbw) [inline src/components/AddressForm/AddressForm.tsx:63]: remove console.log

File: src/components/AddressForm/AddressForm.tsx [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+        url = '/api/posts/create_nova_poshta_warehouse_and_associate_with_post'
+
+        break
+      case 'np_parcel_locker':
+        dataResout = {
+          city: data.city_np_parcel_locker,
+          address_warehouse: data.box_np_parcel_locker
+        }
+        console.log('✌️dataResoutParcelLocer --->', dataResout)
```
Comment (pqoqubbw) [inline src/components/AddressForm/AddressForm.tsx:72]: same

File: src/components/AddressForm/AddressForm.tsx [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+        dataResout = {
+          street: data.street_np_address,
+          house_number: data.house_np_address,
+          apartment_number: data.apartment_np_address,
+          floor: 0,
+          city: data.city_np_address,
+          region: '',
+          area: ''
+        }
+        console.log('✌️dataResoutAddress --->', dataResout)
```
Comment (pqoqubbw) [inline src/components/AddressForm/AddressForm.tsx:85]: same

File: src/components/AddressForm/AddressForm.tsx [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+        dataResout = {
+          street: data.street_urk,
+          house_number: data.house_urk,
+          apartment_number: data.apartment_urk,
+          city: data.city_urk,
+          region: data.region_urk,
+          country: data.country_urk,
+          post_code: data.postalCode_urk
+        }
+        console.log('✌️dataResoutUkrPost --->', dataResout)
```
Comment (pqoqubbw) [inline src/components/AddressForm/AddressForm.tsx:100]: same

File: src/components/AddressForm/AddressForm.tsx [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+        url = '/api/posts/create_ukr_poshta_and_associate_with_post'
+
+        break
+      default:
+        break
+    }
+    axiosInstance
+      .post(url, dataResout)
+      .then((response) => {
+        console.log('Server response:', response)
```
Comment (pqoqubbw) [inline src/components/AddressForm/AddressForm.tsx:110]: same

File: src/components/AddressForm/ErrorMessage.tsx [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+type Errors = FieldErrors<FormValue> | null
+
+type ErrorProps = {
+  errors: Errors
+  fields: Array<keyof FormValue>
+}
+
+export const ErrorMessages: React.FC<ErrorProps> = ({ errors, fields }) => {
+  if (errors) {
+    for (const field of fields) {
```
Comment (pqoqubbw) [inline src/components/AddressForm/ErrorMessage.tsx]: don't use for of

File: src/components/AddressForm/FormRadioGroup.tsx [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+  register: UseFormRegister<FieldValues>
+}
+
+type FormRadioGroupProps = {
+  selectedValue: string
+  setSelectedValue: Dispatch<SetStateAction<string>>
+  renderFormFields: () => ReactNode
+}
+
+function BpRadio(props: any) {
```
Comment (pqoqubbw) [inline src/components/AddressForm/FormRadioGroup.tsx]: don't use any

File: src/components/Modal-custom-btn/ModalCustomBtnAddAddress.tsx [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+import InfoConfirmationModal from './ModalCustomWindow'
+import { stBtn, stTypographyBody2 } from '../Account-panel/style'
+import AddressForm from '../AddressForm/AddressForm'
+import React, { useState } from 'react'
+import AddIcon from '@mui/icons-material/Add'
+import { ISnackbarData } from '../SnackebarCustom/SnackbarCustom'
+
+type ModalCustomBtnAddAddressProps = {
+  setDeliveryAddresses: React.Dispatch<
+    React.SetStateAction<{ nova_poshta: never[]; ukr_poshta: never[] }>
```
Comment (pqoqubbw) [inline src/components/Modal-custom-btn/ModalCustomBtnAddAddress.tsx]: same here

File: src/components/auth/validation.ts [hunk 17]
```diff
@@ -71,3 +71,53 @@ export const ChangeDataSchema = Yup.object().shape({
...
 })
+
+export const AddAddressSchema = (selectedValue: string) => {
+  const commonSchema = Yup.object().shape({
+    country_urk: Yup.string(),
+    apartment_np_address: Yup.string(),
+    apartment_urk: Yup.string()
+  })
+
+  let specificSchema
```
Comment (pqoqubbw) [inline src/components/auth/validation.ts]: don't use let

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/components/Account-panel/Delivery-address/Delivery-address.tsx (hunk 4 above)
Comment (San88Francisco) [reply src/components/Account-panel/Delivery-address/Delivery-address.tsx]: done

File: src/components/Account-panel/Delivery-address/Delivery-address.tsx (hunk 2 above)
Comment (San88Francisco) [reply src/components/Account-panel/Delivery-address/Delivery-address.tsx]: done

File: src/components/Account-panel/Delivery-address/Delivery-address.tsx (hunk 3 above)
Comment (San88Francisco) [reply src/components/Account-panel/Delivery-address/Delivery-address.tsx]: done

File: src/components/Account-panel/Delivery-address/Delivery-address.tsx (hunk 5 above)
Comment (San88Francisco) [reply src/components/Account-panel/Delivery-address/Delivery-address.tsx]: Here I leave only 2 arrays that I work with, for more comfortable work further with it. In novaPoshtaArray, I add the source and addressType fields, this helps me in the further implementation of the logic to iterate over 'відділення' : 'адреса'

File: src/components/AddressForm/ErrorMessage.tsx [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+    const errorMessage = get(errors, field)?.message
+    return errorMessage
+  })
+
+  if (errorField) {
+    const errorMessage = get(errors, errorField)?.message
+    return <p style={{ color: 'red' }}>{errorMessage}</p>
+  }
+
+  return null
```
Comment (pqoqubbw) [inline src/components/AddressForm/ErrorMessage.tsx]:
```suggestion
  if (!errors) return null

  const errorField = fields.find((field) => {
    const errorMessage = get(errors, field)?.message
    return errorMessage
  })
  
  if (!errorField) return null

  const errorMessage = get(errors, errorField)?.message
  return <p style={{ color: 'red' }}>{errorMessage}</p>
```

**Review by pqoqubbw — CHANGES_REQUESTED** [review]

File: src/components/AddressForm/ErrorMessage.tsx (hunk 18 above)
Comment (San88Francisco) [reply src/components/AddressForm/ErrorMessage.tsx]: done

File: src/components/AddressForm/FormRadioGroup.tsx (hunk 15 above)
Comment (San88Francisco) [reply src/components/AddressForm/FormRadioGroup.tsx]: done

File: src/components/AddressForm/AddressForm.tsx (hunk 8 above)
Comment (San88Francisco) [reply src/components/AddressForm/AddressForm.tsx]: done

File: src/components/AddressForm/ErrorMessage.tsx (hunk 14 above)
Comment (San88Francisco) [reply src/components/AddressForm/ErrorMessage.tsx]: done

File: src/components/auth/validation.ts (hunk 17 above)
Comment (San88Francisco) [reply src/components/auth/validation.ts]: done

File: src/components/Modal-custom-btn/ModalCustomBtnAddAddress.tsx (hunk 16 above)
Comment (San88Francisco) [reply src/components/Modal-custom-btn/ModalCustomBtnAddAddress.tsx]: done

File: src/components/Account-panel/Delivery-address/fatchDataPostOffices.ts (hunk 6 above)
Comment (San88Francisco) [reply src/components/Account-panel/Delivery-address/fatchDataPostOffices.ts]: done

**Pull request closed without merging**